content/blog/integrating-k8sgpt-to-empower-kubernetes-with-ai-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,415 @@
+Running on local URL:  http://0.0.0.0:7860
+```
+
+The OpenAI compatible API endpoint is hosted on *'http://0.0.0.0.5000'*. 
+
+Apart from this API endpoint URL, the TGW starts running another local web URL, i.e., at *'http://0.0.0.0:7860'*, providing 3 interface modes, *chat*, *default* & *notebook*, for text generation with the local LLM model backend. 
+
+You can start the browser and type this local URL *'http://0.0.0.0:7860'*. You then land to the *Chat* page. Try to ask some question in the *Chat* page by typing some text and clicking **Generate** button. You may notice it’s a bit slower if you serve the model using CPU inference. But everything should work and you will get the response to your question by *AI*.
+
+
+
+
+
+
+![](/img/local-llm.png)
+
+
+### Integrate K8sGPT with Local AI
+
+
+
+Type the following command to configure K8sGPT to use the local API endpoint. Don’t forget to add the *“/v1”* to the API URL. Hit *Enter* when asking for entering OpenAI key.
+
+
+
+```shell
+$ k8sgpt auth add --backend openai --model Llama-2-13b-chat-hf --baseurl http://localhost:5000/v1
+Enter openai Key: 
+openai added to the AI backend provider list
+```
+
+In case K8sGPT has been already added an OpenAI API endpoint, type below command to remove it. Then re-run the *k8sgpt auth add* to add the new API endpint.
+
+```shell
+$ k8sgpt auth remove openai
+```
+
+
+
+### Detect K8s issues using K8sGPT
+
+
+
+
+With all components being installed and configured, it's time to detect K8s issues in the K8s cluster. 
+
+#### Deploy a sample application
+
+This section will deploy a sample application using *kubectl* CLI.
+
+Type below commands to deploy such a sample application to the namespace *'cfe-apps'*:
+
+```shell
+$ kubectl create ns cfe-apps
+namespace/cfe-apps created
+
+$ kubectl create deployment app-with-no-image --image=cfe-image-not-exist -n cfe-apps
+deployment.apps/app-with-no-image created
+```
+
+Check the deployed application using the following command:
+
+```shell
+$ kubectl get all -n cfe-apps
+NAME                                     READY   STATUS             RESTARTS   AGE
+pod/app-with-no-image-7ff65f5484-9bt4z   0/1     ImagePullBackOff   0           2m
+
+NAME                                READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/app-with-no-image   0/1     1            0            2m
+
+NAME                                           DESIRED   CURRENT   READY   AGE
+replicaset.apps/app-with-no-image-7ff65f5484   1         1         0        2m
+```
+
+The output shows some issues in the K8s *Pod* *'app-with-no-image-7ff65f5484-9bt4z'*. The K8s Deployment keeps showing as *'0/1 READY'*.
+
+#### Run *k8sgpt analyze*
+
+To start K8s issue detection, type the following *k8sgpt analyze* command:
+
+```shell
+$ k8sgpt analyze --filter=Pod --namespace cfe-apps --no-cache
+AI Provider: openai

```

Comment:
I am confused here...AI provider --> shouldn't it be "LocalAI".
Shouldn't you run the analysis using the backend localai (--backend localai)?

Reply:
It's been re-configured with --backend localai.